scripts/run_lighthouse_tests.py

```diff
@@ -50,11 +62,143 @@ def run_lighthouse_checks():
+        common.OPPIA_TOOLS_DIR, 'nginx.conf')
+    with python_utils.open_file(nginx_conf_file, 'w') as f:
+        f.write("""
+            events {
+
+            }
+            http {
+                server {
+                    listen 9999;
+
+                    gzip                on;
+                    gzip_buffers        16 8k;
+                    gzip_comp_level     9;
+                    gzip_http_version   1.1;
+                    gzip_proxied        any;
+                    gzip_types          *;
+                    gzip_vary           on;
+
+                    location / {
+                        proxy_set_header x-real-IP $remote_addr;
+                        proxy_set_header x-forwarded-for $proxy_add_x_forwarded_for;
+                        proxy_set_header host $host;
+                        proxy_pass http://127.0.0.1:8181;
+                    }
+                }
+            }
+        """)
+
+    python_utils.PRINT('Starting proxy server...')
+    start_server_command = ['sudo', 'nginx', '-c', nginx_conf_file]
+    p = subprocess.Popen(start_server_command)
+
+    RUNNING_PROCESSES.append(p)
+
+
+def run_lighthouse_checks_with_compression():
+    """Run lighthouse checks with compression enabled."""
+    # Check if nginx is installed.
+    try:
+        python_utils.PRINT('Checking if nginx is installed...')
+        check_nginx_command = ['which', 'nginx']
+        subprocess.check_call(check_nginx_command)
+    except subprocess.CalledProcessError:
+        download_and_install_nginx()
+
+    constants_env_variable = '"DEV_MODE": false'
+    for line in fileinput.input(
+            files=[os.path.join('assets', 'constants.ts')], inplace=True):
+        # Inside this loop the STDOUT will be redirected to the file,
+        # constants.ts. The end='' is needed to avoid double line breaks.
+        python_utils.PRINT(
+            re.sub(
+                r'"DEV_MODE": .*', constants_env_variable, line), end='')
+    build.main(args=['--prod_env'])
+    start_google_app_engine_server(True)
+    wait_for_port_to_be_open(APP_ENGINE_PORT)
+    start_proxy_server()
+    wait_for_port_to_be_open(NGINX_PORT)
+    run_lighthouse_checks()
+
+
+def cleanup():
+    """Kill the running subprocesses and server fired in this program."""
+    dev_appserver_path = '%s/dev_appserver.py' % common.GOOGLE_APP_ENGINE_HOME
+    processes_to_kill = [
+        '.*%s.*' % re.escape(dev_appserver_path),
+        '.*%s.*' % re.escape('nginx')
+    ]
+
+    for p in RUNNING_PROCESSES:
+        p.kill()
+
+    for p in processes_to_kill:
+        common.kill_processes_based_on_regex(p)
+
+
+def main(args=None):
     """Runs lighthouse checks and deletes reports."""
+    parsed_args = _PARSER.parse_args(args=args)
+    atexit.register(cleanup)
     setup_and_install_dependencies()
-    run_lighthouse_checks()
-    delete_reports()
+    if parsed_args.enable_compression:
+        run_lighthouse_checks_with_compression()
```

Comment:
Why do we have this as an option? I would assume that in 99% cases we would run it with compression.

Reply:
This is for users that are not using Ubuntu and want to run it locally.